ts/src/config.ts

```diff
@@ -77,9 +78,25 @@ export interface Config extends AuthenticationConfig {
   // stack depth may increase overhead of profiling.
   heapMaxStackDepth?: number;
 
-  // Time to wait before trying to create a profile again if profile creation
-  // fails.
-  backoffMillis?: number;
+  // On first error during profile creation, if the backoff is not specified
+  // by the server response, then profiler will between 0 and
+  // initialBackoffMillis before asking the server to create a profile again.
+  // After a successful profile creation, the backoff will be reset to
+  // initialExpBackoffMillis.
+  initialBackoffMillis?: number;
+
+  // If the backoff is not specified by the server response, then profiler will
+  // wait at most maxBackoffMillis before asking server to create a profile
+  // again.
+  maxBackoffMillis?: number;
+
+  // On each consecutive error in profile creation, the maximum backoff will
+  // increase by this factor. The backoff will be random value selected
+  // from a uniform distribution between 0 and the maximum backoff.
```

Comment:
Same.

Reply:
Switched to using "backoff envelope" here.